light-clients/icsxx-cf-solana/src/header.rs

```diff
@@ -1,2 +1,397 @@
-super::wrap!(cf_guest_upstream::Header<PK> as Header);
-super::wrap!(impl<PK> proto for Header);
+use crate::{
+	proto,
+	solana::{
+		blockstore::{get_completed_data_ranges, get_slot_entries_in_block},
+		shred::{shred_code::ShredCode, shred_data::ShredData, *},
+	},
+};
+use alloc::vec::Vec;
+use guestchain::PubKey;
+use proto_utils::BadMessage;
+use solana_sdk::{clock::Slot, hash::Hash, signature::Signature};
+use std::{collections::BTreeSet, convert::From, marker::PhantomData};
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Header<PK> {
+	pub shreds: Vec<Shred>,
```

Comment:
Could we live with this not being `pub` and having read-only accessor?

Reply:
Will create getters

Reply:
The field has type `PreCheckedShreds` now, which is read-only